src/function/table/table_scan.cpp

```diff
@@ -66,10 +66,14 @@ static unique_ptr<FunctionOperatorData> TableScanParallelInit(ClientContext &con
 
 static void TableScanFunc(ClientContext &context, const FunctionData *bind_data_p, FunctionOperatorData *operator_state,
                           DataChunk &output) {
-	auto &bind_data = (const TableScanBindData &)*bind_data_p;
+	auto &bind_data = (TableScanBindData &)*bind_data_p;
 	auto &state = (TableScanOperatorData &)*operator_state;
 	auto &transaction = Transaction::GetTransaction(context);
 	bind_data.table->storage->Scan(transaction, output, state.scan_state, state.column_ids);
+	{
+		lock_guard<mutex> read_lock(bind_data.mutex);
```

Comment:
The mutex only seems to be used to increment `bind_data.chunk_count`, perhaps better to use `atomic<idx_t>`?